breadbox/breadbox/crud/dataset.py

```diff
@@ -1391,68 +1393,81 @@ def get_dataset_feature_by_given_id(
         )
         .one_or_none()
     )
+
     if feature is None:
         raise ResourceNotFoundError(
             f"Feature given ID '{feature_given_id}' not found in dataset '{dataset_id}'."
         )
     return feature
 
 
+def get_dataset_sample_by_given_id(
+    db: SessionWithUser, dataset_id: str, sample_given_id: str,
+) -> DatasetSample:
+    dataset = get_dataset(db, db.user, dataset_id)
+    if dataset is None:
+        raise ResourceNotFoundError(f"Dataset '{dataset_id}' not found.")
+    assert_user_has_access_to_dataset(dataset, db.user)
+    assert isinstance(dataset, MatrixDataset)
+
+    sample = (
+        db.query(DatasetSample)
+        .filter(
+            DatasetSample.given_id == sample_given_id,
+            DatasetSample.dataset_id == dataset.id,
+        )
+        .one_or_none()
+    )
+
+    if sample is None:
+        raise ResourceNotFoundError(
+            f"Sample given ID '{sample_given_id}' not found in dataset '{dataset_id}'."
+        )
+    return sample
+
+
 def get_dataset_feature_by_label(
-    db: SessionWithUser, user: str, dataset_id: str, feature_label: str
+    db: SessionWithUser, dataset_id: str, feature_label: str
 ) -> DatasetFeature:
     """Load the dataset feature corresponding to the given dataset ID and feature label"""
 
-    dataset = get_dataset(db, user, dataset_id)
+    dataset = get_dataset(db, db.user, dataset_id)
     if dataset is None:
         raise ResourceNotFoundError(f"Dataset '{dataset_id}' not found.")
-    assert_user_has_access_to_dataset(dataset, user)
+    assert_user_has_access_to_dataset(dataset, db.user)
     assert isinstance(dataset, MatrixDataset)
 
-    # check for metadata table associated with the features
-    feature_metadata_dataset_id = None
-    if dataset.feature_type is not None:
-        feature_metadata_dataset_id = dataset.feature_type.dataset_id
-
-    if feature_metadata_dataset_id is not None:
-        # look up the given_id from the metadata
-        assert feature_metadata_dataset_id
-        result = (
-            db.query(TabularColumn)
-            .join(TabularCell)
-            .filter(
-                TabularColumn.dataset_id == feature_metadata_dataset_id,
-                TabularColumn.given_id == "label",
-                TabularCell.value == feature_label,
-            )
-            .with_entities(TabularCell.dimension_given_id)
-            .one_or_none()
+    labels_by_given_id = get_dataset_feature_labels_by_id(db, db.user, dataset)
```

Comment:
Has potential to be a performance issue, but I'm content to leave it until it is. ;)

Reply:
Yeah I agree. I think in general, the lookups by label have iffy performance right now. 

If we do run into problems loading a lot of dimensions, we can either:
* look them up by ID instead of label (which I think we're trying to move towards anyway) 
* or improve the performance here